src/main/webapp/ChapMap.js

```diff
@@ -13,43 +13,136 @@
 // limitations under the License.
 
 class ChapMap {
+
+  static SELECTED_CLASS = "selected";
+
   /** The google.maps.Map that will be displayed on the page. */
   googleMap_;
 
   /** The temporary marker visible on this map */
   tempMarker_;
 
-  /** The info window used for all markers*/
+  /** The info window used for all markers */
   myInfoWindow_;
 
-  /** all permanent markers on the page */
-  permMarkers_;
+  /** PermMarker currently being edited */
+  editedPermMarker_;
 
-  /** state determining whether the client can add markers or not */
+  /** State determining whether the client can add markers or not */
   addingMarkers_;
 
-  /** the map's click listener for adding markers */
-  mapClickListener_;
+  constructor() {
+    this.googleMap_ = new google.maps.Map(document.getElementById("map"), {
+        center: { lat: -34.397, lng: 150.644 },
+        zoom: 8
+      });
+
+    this.makeMapButtons_();
+    this.addMapClickListener_();
+
+    this.myInfoWindow_ = new PermInfoWindow();
+    this.tempMarker_ = new TempMarker();
+    this.addingMarkers_ = false;
+    this.editedPermMarker_ = null;
+  }
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// LISTENERS
 
   /**
    * @Private
    * Adds a click listener allowing clients to add markers to the map
    */
   addMapClickListener_() {
-    this.mapClickListener_ = this.googleMap_.addListener('click', (e) => {
-      var coords = e.latLng;
-      this.myInfoWindow_.close();
-      this.tempMarker_.setTempMarker(coords);
-      this.googleMap_.panTo(coords);
+    this.googleMap_.addListener('click', (e) => {
+      if (this.addingMarkers_) {
+        this.editedPermMarker_ = null;
+        this.setTempMarker(e.latLng);
+        this.closeInfoWindow();
+      }
     });
   }
 
   /**
    * @Private
-   * Removes the click listener allowing clients to add markers from the map
+   * Adds click listeners to the map customization buttons.
+   * A client can only add markers when the "adding markers" mode is on
+   */
+  addBtnListeners_(viewBtn, addMarkerBtn) {
+    addMarkerBtn.addEventListener('click', () => this.enableAddingMarkers_());
```

Comment:
I like what you did here, with moving the methods to the class and calling them from an inline function.

Reply:
Thank you! It's definitely a lot cleaner this way.